plugin/pkg/scheduler/algorithm/types.go

```diff
@@ -39,7 +39,8 @@ type PriorityMapFunction func(pod *api.Pod, meta interface{}, nodeInfo *schedule
 type PriorityReduceFunction func(pod *api.Pod, result schedulerapi.HostPriorityList) error
 
 // MetdataProducer is a function that computes metadata for a given pod.
-type MetadataProducer func(pod *api.Pod) interface{}
+// In some cases NodeInfo is required in order for certain types of metadata caching, but leaving nil may be ok in many cases also.
```

Comment:
True, but remember the odd thing is that this interface is used for different factorIes. I guess I can directly say that in the comment rather